plugins/action_update_review_status/src/updateReviewsStatuses.ts

```diff
@@ -18,38 +18,96 @@ const updateReviewsStatuses: ActionPluginType = async ({
   const applicationId = parameters?.applicationId ?? applicationData?.applicationId
+  const reviewId = parameters?.reviewId ?? applicationData?.reviewData?.reviewId
+  const decision = applicationData?.reviewData?.latestDecision?.decision || Decision.NoDecision
 
-  const { changedApplicationResponses = [] } = parameters
+  const isReview = !!reviewId
+  const changedResponses = parameters.changedResponses || []
   const stageId = parameters?.stageId || applicationData?.stageId
-  const currentReviewLevel = applicationData?.reviewData?.levelNumber || 0
-  const level = parameters?.level || currentReviewLevel + 1
+  const currentReviewLevel = parameters.level || applicationData?.reviewData?.levelNumber || 0
+
+  if (isReview) console.log('Review submitted id', reviewId, 'level ', currentReviewLevel)
+  else console.log('Application submitted id', applicationId)
+
+  console.log('Changed responses', changedResponses)
 
   const reviewsToUpdate = []
 
-  console.log('changedApplicationResponses', changedApplicationResponses)
+  const getReviewsByLevelAndStatus = async (
+    level: number,
+    statusToUpdate: ReviewStatus[]
+  ): Promise<Review[]> =>
+    (await db.getAssociatedReviews(applicationId, stageId, level)).filter(
+      (review: Review) =>
+        review.reviewId !== reviewId && statusToUpdate.includes(review.reviewStatus)
+    )
+
+  console.log('Finding reviews to update status...')
 
   try {
-    // Get reviews/review assignments (with status) matching application_id & current stage
-    const reviews = (
-      await db.getAssociatedReviews(applicationId, stageId, level)
-    ).filter((review: Review) =>
-      [ReviewStatus.Submitted, ReviewStatus.Locked, ReviewStatus.Draft].includes(
-        review.reviewStatus
-      )
-    )
-    // Deduce which ones should be updated
-    for (const review of reviews) {
-      const { reviewAssignmentId, levelNumber, reviewStatus } = review
-      if (levelNumber > 1) reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (await haveAssignedResponsesChanged(reviewAssignmentId))
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
-      else if (reviewStatus === ReviewStatus.Locked)
-        reviewsToUpdate.push({ ...review, reviewStatus: ReviewStatus.Pending })
+    if (isReview) {
+      // Review submitted from upper level to lower level review
+      if (decision === Decision.ChangesRequested) {
+        const previousLevelReview = currentReviewLevel - 1
+        // Update lower level reviews with assigned responses on the array changed responses ...
+        const reviewsSubmitted = await getReviewsByLevelAndStatus(previousLevelReview, [
+          ReviewStatus.Submitted,
+        ])
+        // Deduce which ones should be updated to Pending
+        for (const review of reviewsSubmitted) {
+          const { reviewAssignmentId } = review
+          if (await haveAssignedResponsesChanged(reviewAssignmentId))
```

Comment:
There is a [diagram](https://app.diagrams.net/#G1VIyFc7NjAEShXOQxgJoOvKgKrxbtsY0q]), but it doesn't quite describe this in enough details. We should also check that change review response is of status 'disagree'

<img width="687" alt="Screen Shot 2021-05-21 at 12 16 00 PM" src="https://user-images.githubusercontent.com/26194949/119064102-b140ad80-ba2e-11eb-8a2f-250004b6cda1.png">

Reply:
Yes, I was very confused by this diagram. Couldn't really follow for what was required to update reviews:
- From lower level to upper level
- From upper level to lower level

I think the idea is to abstract and the flow is not very intuitive - at least not when I was reading on my own

Reply:
Hmmm I think this will require a bigger change, the `changedResponses` need to pass each decision to this action. And this is one array passed as `outputCumulative` from a previous method. I'll ask @CarlosNZ how is the best way to update that.

Reply:
I guess the logic is not very simple, In this case the idea is that when there is a change request submitted, we change status of change requested for submitted lvl -1 review where
* at least one `new` consolidation review response is disagree for template element that's assigned to lvl -1 reviewer

Reply:
new = updatedResponse

Reply:
yeah can do a better job on the diagram, i can make one from the updateReviewsStatuses logic, once it's finalised

Reply:
Done: 6d7bd4b